packages/demo/src/App.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint react/jsx-key: off */
+import * as React from 'react';
+import { Admin, Resource, CustomRoutes, AuthProvider } from 'react-admin'; // eslint-disable-line import/no-unresolved
+import { Route } from 'react-router-dom';
+
+import comments from './comments';
+import CustomRouteLayout from './customRouteLayout';
+import CustomRouteNoLayout from './customRouteNoLayout';
+import dataProvider from './dataProvider';
+import i18nProvider from './i18nProvider';
+import Layout from './Layout';
+import posts from './posts';
+import users from './users';
+import tags from './tags';
+import Keycloak, { KeycloakConfig } from 'keycloak-js';
+import { keycloakAuthProvider } from 'ra-keycloak';
+
+const App = () => {
+    const [authProvider, setAuthProvider] = React.useState<AuthProvider | null>(
+        null
+    );
+
+    React.useEffect(() => {
+        async function startAuthProvider() {
+            const config: KeycloakConfig = {
+                url: 'http://localhost:8080/auth',
+                realm: 'Marmelab',
+                clientId: 'front-marmelab',
+            };
+
+            const keycloak = new Keycloak(config);
+            await keycloak.init({ onLoad: 'login-required' });
+            const authProvider = keycloakAuthProvider(keycloak, {});
+            setAuthProvider(authProvider);
+        }
+        if (authProvider === null) {
+            startAuthProvider();
+        }
+    }, [authProvider]);
+
+    // hide the admin until the data provider is ready
+    if (!authProvider) return <p>Loading...</p>;
+
+    return (
+        <Admin
+            authProvider={authProvider}
+            dataProvider={dataProvider}
```

Comment:
In the end you do not use the custom `httpClient` in your dataProvider, do you? I think we should do it to demonstrate it as well